src/map/skill.cpp

```diff
@@ -14512,21 +14513,24 @@ int32 skill_castend_pos2(struct block_list* src, int32 x, int32 y, uint16 skill_
 	case SA_VOLCANO:
 	case SA_DELUGE:
 	case SA_VIOLENTGALE:
-	{	//Does not consumes if the skill is already active. [Skotlex]
-		std::shared_ptr<s_skill_unit_group> sg2;
-		if ((sg2= skill_locate_element_field(src)) != nullptr && ( sg2->skill_id == SA_VOLCANO || sg2->skill_id == SA_DELUGE || sg2->skill_id == SA_VIOLENTGALE ))
-		{
-			if (sg2->limit - DIFF_TICK(gettick(), sg2->tick) > 0)
-			{
-				skill_unitsetting(src,skill_id,skill_lv,x,y,0);
-				return 0; // not to consume items
+		// Does not consumes if the skill is already active. [Skotlex]
+		if (std::shared_ptr<s_skill_unit_group> sg2 = skill_locate_element_field(src); sg2 != nullptr) {
+			switch( sg2->skill_id ) {
+				case SA_VOLCANO:
+				case SA_DELUGE:
+				case SA_VIOLENTGALE:
+					if (sg2->limit - DIFF_TICK(gettick(), sg2->tick) > 0) {
+						skill_unitsetting(src,skill_id,skill_lv,x,y,0);
+						return 0; // not to consume items
+					}
+					sg2->limit = 0; // Disable it.
+					break;
 			}
-			else
-				sg2->limit = 0; //Disable it.
 		}
+		sc_start(src, src, SC_GROUNDMAGIC, 100, 0, skill_get_time(skill_id, skill_lv));
```

Comment:
This should only happen if skill_unitsetting managed to place at least one unit.

Reply:
Wouldnt it be better to move this sc_start into skill_unit_effect or another appropriate function?